File: .taira-source-c80bf4d85a97/java/iroha_android/src/main/java/org/hyperledger/iroha/android/crypto/export/KeyExportStore.java
package org.hyperledger.iroha.android.crypto.export;

import java.util.Optional;

/**
 * Persistence backend for deterministic key export bundles.
 *
 * <p>Stores the base64-encoded representation returned by {@link KeyExportBundle#encodeBase64()}.
 */
public interface KeyExportStore {
  /** Loads the stored bundle for {@code alias}, when present. */
  Optional<String> load(String alias) throws KeyExportException;

  /** Stores the base64-encoded bundle for {@code alias}. */
  void store(String alias, String bundleBase64) throws KeyExportException;

  /** Removes any stored bundle for {@code alias}. */
  void delete(String alias) throws KeyExportException;
}

File: .taira-source-c80bf4d85a97/kotlin/core-jvm/src/main/java/org/hyperledger/iroha/sdk/crypto/export/KeyExportStore.kt
package org.hyperledger.iroha.sdk.crypto.export

/**
 * Persistence backend for deterministic key export bundles.
 *
 * Stores the base64-encoded representation returned by `KeyExportBundle.encodeBase64()`.
 */
interface KeyExportStore {
    /** Loads the stored bundle for `alias`, when present. */
    @Throws(KeyExportException::class)
    fun load(alias: String): String?

    /** Stores the base64-encoded bundle for `alias`. */
    @Throws(KeyExportException::class)
    fun store(alias: String, bundleBase64: String)

    /** Removes any stored bundle for `alias`. */
    @Throws(KeyExportException::class)
    fun delete(alias: String)
}
